src/decision/src/GpsDecision.cpp

```diff
@@ -36,29 +30,48 @@ GpsDecision::GpsDecision(int argc, char **argv, std::string node_name) {
     // based on our distance and heading to the destination GPS waypoint
     double linear_heading_factor, linear_distance_factor,
             angular_heading_factor, angular_distance_factor;
-    double linear_cap, angular_cap;
     SB_getParam(private_nh, "linear_heading_factor", linear_heading_factor, 1.0);
     SB_getParam(private_nh, "linear_distance_factor", linear_distance_factor, 1.0);
     SB_getParam(private_nh, "angular_heading_factor", angular_heading_factor, 1.0);
     SB_getParam(private_nh, "angular_distance_factor", angular_distance_factor, 1.0);
     mover.setFactors(linear_distance_factor, linear_heading_factor,
                     angular_distance_factor, angular_heading_factor);
+
+    double linear_cap, angular_cap;
     SB_getParam(private_nh, "max_linear_speed", linear_cap, 1.0);
     SB_getParam(private_nh, "max_angular_speed", angular_cap, 1.0);
     mover.setMaxSpeeds(linear_cap, angular_cap);
-}
-
-void GpsDecision::currentLocationCallback(const geometry_msgs::Point::ConstPtr &current_location) {
-    this->current_location = *current_location;
-}
 
-void GpsDecision::imuCallback(const std_msgs::Float32::ConstPtr &heading) {
-    this->current_heading = heading->data;
+    SB_getParam(private_nh, "base_frame", base_frame, (std::string)"base_link");
+    SB_getParam(private_nh, "global_frame", global_frame, (std::string)"odom_combined");
 }
 
-void GpsDecision::waypointCallback(const geometry_msgs::Point::ConstPtr &waypoint) {
-    // Create a new twist message and publish it
-    geometry_msgs::Twist twist = mover.createTwistMessage(current_location, current_heading, *waypoint);
-    twist_publisher.publish(twist);
+void GpsDecision::waypointCallback(const geometry_msgs::PointStamped::ConstPtr &waypoint) {
+    // Get the current position and heading of the robot
+    tf2_ros::Buffer tfBuffer;
+    tf2_ros::TransformListener tfListener(tfBuffer);
+    try {
+        geometry_msgs::TransformStamped tfStamped = tfBuffer.lookupTransform(
+                global_frame, base_frame, ros::Time(0), ros::Duration(1.0));
```

Comment:
okie, so now I remember why this doesn't work either... since we can't make the `TransformListener` a class member, we publish the waypoint and set the stamp to the current time, but we only start listening for TF's after we get the waypoint, and we can't get TF's from the past, so we crash. On the other hand, when deciding on how to move, we want to use the most up to date TF to determine the current state of the robot anyways, so using `ros::Time(0)` is probably the right choice.